packages/compiler/src/collector/expression.ts

```diff
@@ -37,6 +39,31 @@ export const handleJsxExpression = (node: Node<ts.Node>) => {
           ? undefined
           : arrayExpression;
       })
+      // fontSize={2 + 5}
+      .when(Node.isBinaryExpression, (exp) => {
+        const leftOperand = handleJsxExpression(exp.getLeft());
+        const rightOperand = handleJsxExpression(exp.getRight());
+        const operator = exp.getOperatorToken().getKind();
+        if (
+          typeof leftOperand === "number" &&
+          typeof rightOperand === "number"
+        ) {
+          // FIXME: opted for native switch due to a type conflict between ts-pattern and SyntaxKind which remains unexplained.
```

Comment:
This is a TS limitation.
It would be difficult to resolve type errors with ts-pattern.

https://github.com/gvergnaud/ts-pattern/issues/183

Reply:
I wasn't aware of that! Thank you for letting me know 👍